src/components/MeasurementOverlay/CustomMeasurementOverlay.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withTranslation } from 'react-i18next';
+import Icon from 'components/Icon';
+import core from 'core';
+import { mapAnnotationToKey, getDataWithKey } from '../../constants/map';
+
+function CustomMeasurementOverlay(props) {
+  const renderAppropriateOverlay = type => {
+    const annotationType = mapAnnotationToKey(props.annotation);
+    switch (annotationType) {
+      case 'ellipse':
+        return <CustomEllipseMeasurementOverlay {...props}/>;
+      default:
+        console.error(`Custom overlay for annotation type: ${type} not supported`);
+    }
+  };
```

Comment:
I found that if we end up implementing overlays for other annotations they will have some variation... and it's hard to come up with a base component to cover all cases. Therefore I think this "factory" approach is best, but maybe there is a better way?